cli/boards.go

```diff
@@ -0,0 +1,234 @@
+package cli
+
+import (
+	"archive/tar"
+	"bufio"
+	"bytes"
+	"compress/gzip"
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/urfave/cli/v2"
+	"go.uber.org/multierr"
+	packagepb "go.viam.com/api/app/packages/v1"
+)
+
+// supportedVersionRegex validates that the board version is semver 2.0.0 specification.
+var supportedVersionRegex = regexp.MustCompile(`^(?P<major>0|[1-9]\d*)\.(?P<minor>0|[1-9]\d*)\.(?P<patch>0|[1-9]\d*)` +
+	`(?:-(?P<prerelease>(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)` +
+	`(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+(?P<buildmetadata>[0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?$`)
+
+const boardUploadChunkSize = 32 * 1024
```

Comment:
This isn't really a chunk size, it's the maximum file size of the tarball (post-compression, so it's okay to have raw JSON files larger than this). I'd rename to make that clearer.

Reply:
fine with making it bigger, I used 32kB since that is the chunk size used rn for uploading capture data to the cloud. It is being used as the maximum now since we're only supporting one json file (can change the name to reflect that for now) but when we add support for a package with multiple json files then it can be bigger than 32kB and we will just split it into 32kB chunks for the stream. i'm not sure if doing bigger chunks would cause issues or not.